### ISSUE TITLE
Is there a way to disable auto-imports?

### ISSUE THREAD

**Issue by PedroMDuarte** [microsoft/pylance-release#64]
I find that I would prefer to have that option off by default.

**Comment by savannahostrowski** [comment]
Hey there @PedroMDuarte 

Thanks for the suggestion. This is something that we are currently investigating. Stay tuned!

**Comment by jakebailey** [comment]
Just to give an update here: We're planning on adding a setting to allow auto-import completions to be disabled, as well as modify the tooltips to make it more obvious what text changes the completions are going to make (i.e., the import that will be added to the top).

**Comment by jakebailey** [comment]
This issue has been fixed in version 2020.8.0, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/master/CHANGELOG.md#202080-5-august-2020

**Comment by jakebailey** [comment]
TL;DR, you can now set:

```json
{
    "python.analysis.autoImportCompletions": false
}
```

To disable the auto-import completions if you prefer not to have them.

Note that the quick-fixes will still be available but still behind the lightbulb/"quick-fix" button (both of which VS Code should allow you to customize).

The aforementioned updates to the tooltips are coming, but will be incremental over time as we refine how they're presented.

**Comment by jakebailey** [comment]
The updates to the tooltips are now released in 2020.9.0: https://github.com/microsoft/pylance-release/blob/master/CHANGELOG.md#202090-3-september-2020

For example:

![MicrosoftTeams-image](https://user-images.githubusercontent.com/5341706/92181750-e05eb280-edfe-11ea-8866-0450af800fab.png)

**Comment by LoNicoM** [comment]
can't remember how many times i have had to delete "import imp" from the top of my files.

**Comment by ShowierData9978-Alt** [comment]
it is  the worst lol

**Comment by amine4567** [comment]
This feature is the worst, it shouldn't even exist or at least be disabled by default.

**Comment by danechitoaie** [comment]
I've been looking for this setting. It was so annoying. Happy that we can turn it off.

**Comment by frederikheld** [comment]
I would like to keep suggestions while I'm typing the `import ...` line but I totally want to turn off that annoying behavior that makes VSCode write import statements just because it _thinks_ that I need to import some library while I'm writing a line of code somewhere else. Because there was not one single occasion that this was actually when I actually needed that import.

How do I need to tune my settings to achieve that?

**Comment by debonte** [comment]
> I would like to keep suggestions while I'm typing the `import ...` line but I totally want to turn off that annoying behavior that makes VSCode automatically add import statements just because it _thinks_ that I need to import some library while I'm writing a line of code somewhere else. There was not one single occasion where I actually needed that import.
> 
> How do I need to tune my settings to achieve that?

@frederikheld, `python.analysis.autoImportCompletions` is the setting that controls that in Pylance. It's off by default though, which annoys people with the opposite opinion. 😆 If this doesn't help, please file a new issue. This one has been closed for a couple years now. You'll get quicker responses from us generally by filing new issues than commenting on old, closed issues.
